cmd/gopherbadgeimg/main.go

```diff
@@ -0,0 +1,301 @@
+func LoadImg(infile string) (*image.Image, error) {
+	f, err := os.Open(infile)
+	if err != nil {
+		return nil, err
+	}
+	src, _, err := image.Decode(f)
+	if err != nil {
+		return nil, err
+	}
+	return &src, nil
+}
+
+// ImgToBytes resizes an image to the requested size and converts it to a bitmap byte slice
+func ImgToBytes(x, y int, inputImg *image.Image) []byte {
+	// work on values not pointers
+	src := *inputImg
+	// create a new, rectangular image that's the size we want
+	dst := image.NewRGBA(image.Rect(0, 0, x, y))
+	// use NearestNeighbor algo to fit our original image into the smaller (or bigger!?) image
+	draw.NearestNeighbor.Scale(dst, dst.Rect, src, src.Bounds(), draw.Over, nil)
+
+	// Our e-ink display uses one bit for each pixel, on or off.
+	// Therefore, we need one bit for each pixel.
+	// Since we have a byte slice, and 8 bytes per bit, divide by 8
+	imageBits := make([]byte, x*y/8)
+
+	// Again, on or off, white or black are our only color options
+	palette := []color.Color{
+		color.Black,
+		color.White,
+	}
+
+	if disableDithering {
+		// don't dither image if flag is set, useful for some images which are already black and white
+	} else {
+		// using our palette, create a dithering struct
+		// and dither our image to get some false shading.
+		// read more here: https://en.wikipedia.org/wiki/Floyd%E2%80%93Steinberg_dithering
+		d := dither.NewDitherer(palette)
+		d.Matrix = dither.FloydSteinberg
+		dithered := d.Dither(dst)
+		// this nil check is necessary since the library will often write
+		// the dithered image to dst, but not always. Read their docs for more info
+		if dithered != nil {
+			var ok bool
+			dst, ok = dithered.(*image.RGBA)
+			// docs claim image is guaranteed to be of this type when not nil, but it's good to check anyway
+			if !ok {
+				log.Fatalf("error: typeof dithered should have been `*image.RGBA` but was `%T`", dithered)
+			}
+		}
+	}
+
+	// loop over the x axis first, then y as screen updates LTR, top to bottom
+	// (vertical axis must be inner loop) for the badge layout
+	for i := 0; i < x; i++ {
+		for j := 0; j < y; j++ {
+			// grab dithered image point, determine if bit should be 1 or a 0
+			r, g, b, _ := dst.At(x-1-i, j).RGBA()
+			if r+g+b == 0 {
+				// use bit shifting + integer division & modulo arithmetic to change
+				// the individual bits we want to set
+				//			imageBits[(i*y+j)/8] = imageBits[(i*y+j)/8] | (1 << uint(7-(i*y+j)%8))
+				index := (i*y + j) / 8
+				bit := uint(7 - (i*y+j)%8)
+				imageBits[index] |= (1 << bit)
+			}
+		}
+	}
+	return imageBits
+}
+
+func ParseRatio(rstr string) (int, int, error) {
+	rstr = strings.ToLower(rstr)
+	pixels := strings.Split(rstr, "x")
+	if len(pixels) != 2 {
+		return 0, 0, errors.New("invalid ratio string provided")
+	}
+	x, err := strconv.Atoi(pixels[0])
+	if err != nil {
+		return 0, 0, errors.Join(errors.New("error: could not parse x coordinate count"), err)
+	}
+	y, err := strconv.Atoi(pixels[0])
```

Comment:
this should be 
```y, err := strconv.Atoi(pixels[1])```

if not, it takes for both (x,y) only the first part of the ratio **AAA**xBBB

Reply:
ah, right you are, typo.